libconsensus/pbft/PBFTSealer.cpp

```diff
@@ -97,5 +97,99 @@ void PBFTSealer::stop()
+{
+    /// if is syncing, return directly
+    if (m_blockSync->isSyncing())
+    {
+        m_lastBlockNumber = m_blockSync->status().knownHighestNumber;
+        return;
+    }
+    m_timeoutCount++;
+    /// update the last minimum transaction number that lead to timeout
+    if (sealingTxNumber > 0 && (m_lastTimeoutTx == 0 || (m_lastTimeoutTx > sealingTxNumber &&
+                                                            sealingTxNumber > m_maxNoTimeoutTx)))
+    {
+        m_lastTimeoutTx = sealingTxNumber;
+    }
+    else
+    {
+        m_lastTimeoutTx = m_maxNoTimeoutTx;
+    }
+    /// update the maxBlockCanSeal
+    {
+        UpgradableGuard l(x_maxBlockCanSeal);
+        if (m_maxBlockCanSeal > 2)
+        {
+            UpgradeGuard ul(l);
+            m_maxBlockCanSeal /= 2;
+        }
+    }
+    PBFTSEALER_LOG(DEBUG) << LOG_DESC("decrease maxBlockCanSeal to half for PBFT timeout")
+                          << LOG_KV("org_maxBlockCanSeal", m_maxBlockCanSeal * 2)
+                          << LOG_KV("halfed_maxBlockCanSeal", m_maxBlockCanSeal)
+                          << LOG_KV("timeoutCount", m_timeoutCount)
+                          << LOG_KV("lastTimeoutTx", m_lastTimeoutTx);
+}
+
+/// increase maxBlockCanSeal when commitBlock with no-timeout
+void PBFTSealer::onCommitBlock(
+    uint64_t const& blockNumber, uint64_t const& sealingTxNumber, unsigned const& changeCycle)
+{
+    /// if is syncing or timeout, return directly
+    if (m_blockSync->isSyncing() || changeCycle > 0)
+    {
+        m_lastBlockNumber = m_blockSync->status().knownHighestNumber;
+        return;
+    }
+    if (blockNumber <= m_lastBlockNumber)
+    {
+        return;
+    }
+    m_lastBlockNumber = m_blockChain->number();
+    /// update the maximum number of transactions that has been consensused without timeout
+    if (sealingTxNumber > 0 && (m_maxNoTimeoutTx == 0 || m_maxNoTimeoutTx < sealingTxNumber))
+    {
+        m_maxNoTimeoutTx = sealingTxNumber;
+        PBFTSEALER_LOG(DEBUG) << LOG_DESC("increase maxNoTimeoutTx")
+                              << LOG_KV("maxNoTimeoutTx", m_maxNoTimeoutTx);
+    }
+    /// sealing more transactions when no timeout
+    if (m_timeoutCount > 0)
+    {
+        m_timeoutCount--;
+        return;
+    }
+    /// only if m_timeoutCount decreases to 0 and the current maxBlockCanSeal is smaller than
+    /// maxBlockTransactions, increase maxBlockCanSeal
+    if (maxBlockCanSeal() >= m_pbftEngine->maxBlockTransactions())
+    {
+        return;
+    }
+    /// if the current maxBlockCanSeal is larger than m_lastTimeoutTx, return directly
+    if (m_lastTimeoutTx != 0 && maxBlockCanSeal() >= m_lastTimeoutTx)
+    {
+        return;
+    }
+    /// increase the maxBlockCanSeal when its current value is smaller than the
+    /// last-timeout-tx-number
+    {
+        WriteGuard l(x_maxBlockCanSeal);
+        m_maxBlockCanSeal += (0.5 * m_maxBlockCanSeal);

```

Comment:
make the 0.5 can be change in config.ini

Reply:
already add `enable_dynamic_block_size` and `block_size_increase_ratio` to control the increase speed, please refer to https://github.com/FISCO-BCOS/FISCO-BCOS/pull/695/files#diff-708d2961ab288bc22804247af5e4c497R192